apps/aecore/src/aec_blocks.erl

```diff
@@ -104,15 +92,16 @@ new(LastBlock, Txs, Trees0) ->
 
     {ok, Txs1, Trees} = aec_tx:apply_signed(Txs, Trees0, Height),
     {ok, TxsRootHash} = aec_txs_trees:root_hash(aec_txs_trees:from_txs(Txs1)),
-    #block{height = Height,
-           prev_hash = LastBlockHeaderHash,
-           root_hash = aec_trees:hash(Trees),
-           trees = Trees,
-           txs_hash = TxsRootHash,
-           txs = Txs1,
-           target = target(LastBlock),
-           time = aeu_time:now_in_msecs(),
-           version = ?CURRENT_BLOCK_VERSION}.
+    NewBlock =
```

Comment:
I would rename new/3 to new_with_state/3 and remove the {ok, _, _} and only return a 2-tuple.

New is no longer a desciptive name. If we want a function that don't care about the state, we can add a new/3 as well.

Reply:
Targeted by #513 (WIP).